### PR TITLE
wasm-tools: init at 1.0.27

### PR DESCRIPTION
**Pull request by ereslibre**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/wasm-tools/add-Cargo.lock.patch [hunk 1]
```diff
@@ -0,0 +1,2168 @@
+diff --git a/Cargo.lock b/Cargo.lock
```
Comment (figsoda) [inline pkgs/tools/misc/wasm-tools/add-Cargo.lock.patch:1]: please vendor the `Cargo.lock` directly and use

```nix
{
  # ...
  cargoLock = {
    lockFile = ./Cargo.lock;
  };
  # ...
}
```

another option is to use the crate version, which has `Cargo.lock` but doesn't have submodules

```nix
fetchCrate {
  inherit pname version;
  hash = "sha256-2C1HT8B5AEWiv5sDJmXPpAPJ+3qGLYiHJLNmuooNElI=";
}
```

File: pkgs/tools/misc/wasm-tools/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{ rustPlatform
+, fetchFromGitHub
+, lib
+, stdenv
+, Security }:
```
Comment (figsoda) [inline pkgs/tools/misc/wasm-tools/default.nix:5]:
```suggestion
, Security
}:
```

File: pkgs/tools/misc/wasm-tools/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{ rustPlatform
+, fetchFromGitHub
+, lib
+, stdenv
```
Comment (figsoda) [inline pkgs/tools/misc/wasm-tools/default.nix:4]:
```suggestion
{ lib
, rustPlatform
, fetchFromGitHub
, stdenv
```

File: pkgs/tools/misc/wasm-tools/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  src = fetchFromGitHub {
+    owner = "bytecodealliance";
+    repo = pname;
+    rev = "${pname}-${version}";
+    hash = "sha256-kuTcxZLtQyDcj8SFfpJRNwto1e5iuXjxqZ46CnLOVIc=";
+    fetchSubmodules = true;
+  };
+
+  buildInputs = lib.optional stdenv.isDarwin Security;
+
```
Comment (figsoda) [inline pkgs/tools/misc/wasm-tools/default.nix:20]:
```suggestion
```

File: pkgs/tools/misc/wasm-tools/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  cargoLock.lockFile = ./Cargo.lock;
+  postPatch = ''
+    ln -s ${./Cargo.lock} Cargo.lock
+  '';
+
+  cargoHash = "sha256-6UXHBDbObv2HsC99WfhwOaesUF8diLH8Kqgn4dkD2cc=";
```
Comment (figsoda) [inline pkgs/tools/misc/wasm-tools/default.nix:26]:
```suggestion
  buildInputs = lib.optional stdenv.isDarwin Security;
```

**Review by figsoda — COMMENTED** [review]
`

File: pkgs/tools/misc/wasm-tools/default.nix [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+  cargoBuildFlags = [ "--package wasm-tools" ];
```
Comment (figsoda) [inline pkgs/tools/misc/wasm-tools/default.nix:28]:
```suggestion
  cargoBuildFlags = [ "--package" "wasm-tools" ];
```

File: pkgs/tools/misc/wasm-tools/default.nix [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+  cargoTestFlags = [ "--all" ];
+
+  meta = with lib; {
+    description = "Low level tooling for WebAssembly in Rust";
+    homepage = "https://github.com/bytecodealliance/wasm-tools";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ ereslibre ];
+    platforms = platforms.unix;
```
Comment (figsoda) [inline pkgs/tools/misc/wasm-tools/default.nix:37]: We can use `buildRustPackage`'s default here
```suggestion
```

**Review by figsoda — COMMENTED** [review]
`cargoHash` and `cargoLock` should not be used at the same time, and some style nits

File: pkgs/tools/misc/wasm-tools/default.nix (hunk 3 above)
Comment (figsoda) [reply pkgs/tools/misc/wasm-tools/default.nix:4]: > [Arguments should be listed in the order they are used, with the exception of `lib`, which always goes first.](https://nixos.org/manual/nixpkgs/unstable/#sec-syntax)

Comment (ereslibre) [reply pkgs/tools/misc/wasm-tools/default.nix]: Ouch. I am a maintainer of some packages and I just learnt about it, sorry. Updated, thanks!

Comment (figsoda) [reply pkgs/tools/misc/wasm-tools/default.nix]: No problem, this is kind of hidden in blocks of text, understandable

**Review by figsoda — APPROVED** [review]

**Pull request merged**